src/redis_fns.rs

```diff
@@ -60,27 +51,23 @@ pub async fn update_remaining_allowance(
     allowance: u64,
 ) -> Result<u64, RedisError> {
     let mut conn = get_redis_cnxn().await?;
-    let key = account_id.clone().to_string();
-    let gas_used: u64 = (gas_used_in_yn / YN_TO_GAS) as u64;
+    let gas_used: u64 =
+        u64::try_from(gas_used_in_yn / YN_TO_GAS).expect("Gas value must always fit in u64"); // possible truncation
```

Comment:
I don't want the thread to panic if the expect value is not met, just return an error.

Reply:
Got it. Slightly unrelated: on the next line, what happens if the gas usage exceeds the allowance? I didn't directly see a check for this, just [this one](https://github.com/near/pagoda-relayer-rs/blob/c49ec687d463ae23ab712ea3991e1bda1151852a/src/main.rs#L783).